Tests: Bluetooth: fix failing unittests

kruithofa opened the pull request:
Due to some refactoring in the past a few unittests have started failing.
This PR fixes 2 tests for direction finding due to refactoring/new kconfigs
and 1 test for audio codec due to updating to BT spec 5.3

Added Piotr Pryga and Morten Priess since they were involved in the original tests
and/or code changes

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

Commit:
tests: Bluetooth: update unittests for Audio codecs

An LE controller shall no longer support
HCI_Read_Local_Supported_Codecs [v1]
The code in subsys/bluetooth/controller was updated for this
behaviour, and this commit updates the unittest

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

jhedberg commented:
Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.

kruithofa commented:
> Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.

So should I prefix it with 'bluetooth: unittest' instead?

Thalley commented:
> > Why do you use "controller" in the PR and commit message prefixes, even though this doesn't touch any controller code? That's quite confusing. Seems like this is only modifying host and test app code. I was just about to reassign this to @cvinayak when I realised this is actually not a controller PR at all.
> 
> So should I prefix it with 'bluetooth: unittest' instead?

`tests: Bluetooth` is better :)

Thalley commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_EXT_ADV) && defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_EXT_ADV && CONFIG_BT_BROADCASTER */
```
You sure about this? 

`bt_adv_reset_adv_pool` seems to be fine without `CONFIG_BT_EXT_ADV`

```c
void bt_adv_reset_adv_pool(void)
{
#if defined(CONFIG_BT_EXT_ADV)
	(void)memset(&adv_pool, 0, sizeof(adv_pool));
#endif /* defined(CONFIG_BT_EXT_ADV) */

	(void)memset(&bt_dev.adv, 0, sizeof(bt_dev.adv));
}
```

kruithofa replied:
No, I discussed this with Sean and will update

jhedberg commented on subsys/bluetooth/host/hci_core.c:
```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_BROADCASTER */
```
Since this part is fixing something in the host and not a test case, wouldn't it make sense to split it into a separate commit?

kruithofa replied:
In hindsight probably yes, the bug got caught in a unittest though. If a 2nd request for splitting it out comes in I'll put it in a separate commit, otherwise I'll leave it as is

Thalley replied:
I would agree with Johan that it's best to split it. 

I won't block this PR based on that though

kruithofa replied:
I'll split

jhedberg reviewed (commented):
Thanks for updating the prefixes! One additional comment inline.

Commit:
tests: Bluetooth: update unittests for DF

Due to some refactoring the unittests for direction finding
started failing. This commit fixes the failing unittests

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>

Commit:
Bluetooth: host: fix conditional compile for broadcaster role

The call to bt_adv_reset_adv_pool is only required in the
broadcaster role, and not in the observer role, regardless
of the setting of CONFIG_BT_EXT_ADV
As the code was this call was also made for the observer.

Note: handling of the setting of CONFIG_BT_EXT_ADV
is already handled in the bt_adv_reset_adv_pool function

Signed-off-by: Andries Kruithof <andries.kruithof@nordicsemi.no>
